lib/firefox/webdriver/index.js

```diff
@@ -11,10 +11,37 @@ const get = require('lodash.get');
 const geckodriver = require('@sitespeed.io/geckodriver');
 const defaultFirefoxPreferences = require('./firefoxPreferences');
 
+let hasConfiguredGeckoDriverService = false;
+
 module.exports.configureBuilder = function(builder, baseDir, options) {
   const firefoxConfig = options.firefox || {};
   const moduleRootPath = path.resolve(__dirname, '..', '..', '..');
 
+  if (!hasConfiguredGeckoDriverService) {
+    const geckodriverPath = get(
+      firefoxConfig,
+      'geckodriverPath',
+      geckodriver.binPath()
+    );
+
+    let serviceBuilder = new firefox.ServiceBuilder(geckodriverPath);
+    if (options.verbose >= 2) {
+      // This echoes the output from geckodriver to the console.
+      serviceBuilder.setStdio('inherit');
+      // TODO: serviceBuilder.loggingTo(`${baseDir}/geckodriver.log`);
+      if (options.verbose >= 3) serviceBuilder.enableVerboseLogging();
+    }
+    firefox.setDefaultService(serviceBuilder.build());
+
+    // `selenium-webdriver` fixes a port for `geckodriver` when the
+    // service is built, but in between browsertime iterations the
+    // port is not held.  That means the OS can allocate it, and
+    // indeed collisions are frequent in the wild.  This is
+    // counter-intuitive but avoids that port collision by making each
+    // iteration fix a new port for `geckodriver`.
+    hasConfiguredGeckoDriverService = false;
```

Comment:
This variable is always false.  Should this line be setting it to true?

Reply:
Looking at this more closely,  it seems this code is only ever run once per iteration anyways.  Do we even need this variable?

Reply:
I removed the variable and updated the comment.